arcgisUM.py

```diff
@@ -0,0 +1,206 @@
+    
+    if len(courseUsers) == 0:
+        logger.info('No new users to add to ArcGIS Group {}'.format(groupNameAndID))
+        return instructorLog
+
+    logger.info('Adding Canvas Users to ArcGIS Group {}: {}'.format(groupNameAndID, courseUsers))
+    # ArcGIS usernames are U-M uniqnames with the ArcGIS organization name appended.
+    user="NULL"
+    arcGISFormatUsers = formatUsersNamesForArcGIS(user, courseUsers)
+    logger.debug("addCanvasUsersToGroup: formatted: {}".format(arcGISFormatUsers))
+    
+    #results = group.addUsersToGroups(users=','.join(arcGISFormatUsers))
+    results = group.add_users(arcGISFormatUsers)
+    logger.debug("adding: results: {}".format(results))
+
+    usersNotAdded = results.get('notAdded')
+    """:type usersNotAdded: list"""
+    usersCount = len(arcGISFormatUsers)
+    usersCount -= len(usersNotAdded) if usersNotAdded else 0
+    logger.debug("usersCount: {}".format(usersCount))
+    logger.debug("aCUTG: instructorLog 1: [{}]".format(instructorLog))
+    instructorLog += 'Number of users added to group: [{}]\n\n'.format(usersCount)
+    logger.debug("aCUTG: instructorLog 2: [{}]".format(instructorLog))
+    if usersNotAdded:
+        logger.warning('Warning: Some or all users not added to ArcGIS group {}: {}'.format(groupNameAndID, usersNotAdded))
+        instructorLog += 'Users not in group (these users need ArcGIS accounts created for them):\n' + '\n'.join(['* ' + userNotAdded for userNotAdded in usersNotAdded]) + '\n\n' + 'ArcGIS group ID number:\n{}\n\n'.format(group.id)
+    instructorLog += '- - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n'
+    logger.debug("aCUTG: instructorLog 3: [{}]".format(instructorLog))
+
+    logger.info("addCanvasUsersToGroup: instructorLog: [{}]".format(instructorLog))
+    return instructorLog
+
+
+def getCurrentArcGISMembers(group, groupNameAndID):
+    groupAllMembers = {}
+
+    try:
+        groupAllMembers = group.get_members()
+    except RuntimeError as exception:
+        logger.error('Exception while getting users for ArcGIS group "{}": {}'.format(groupNameAndID, exception))
+            
+    groupUsers = groupAllMembers.get('users')
+    """:type groupUsers: list"""
+    return groupUsers
+
+def removeListOfUsersFromArcGISGroup(group, groupNameAndID, groupUsers):
+    """Remove only listed users from ArcGIS group."""
+
+    if len(groupUsers) == 0:
+        logger.info('No obsolete users to remove from ArcGIS Group {}'.format(groupNameAndID))
+        return None
+
+    logger.info('ArcGIS Users to be removed from ArcGIS Group [{}] [{}]'.format(groupNameAndID, ','.join(groupUsers)))
+    results = None
+    try:
+            results = group.removeUsersFromGroup(','.join(groupUsers))
+    except RuntimeError as exception:
+            logger.error('Exception while removing users from ArcGIS group "{}": {}'.format(groupNameAndID, exception))
+            
+    usersNotRemoved = results.get('notRemoved')
```

Comment:
what will happen here in case if an exception is thrown on line 155? I ask because `results` would be `None`

Reply:
It will now return None from the exception.